nft_ingester/src/gapfiller.rs

```diff
@@ -1,9 +1,112 @@
+use crate::sequence_consistent::SequenceConsistentGapfiller;
+use backfill_rpc::rpc::BackfillRPC;
 use futures::StreamExt;
+use grpc::client::Client;
 use interface::asset_streaming_and_discovery::{AssetDetailsConsumer, RawBlocksConsumer};
+use interface::slots_dumper::SlotsDumper;
+use metrics_utils::SequenceConsistentGapfillMetricsConfig;
 use rocks_db::Storage;
 use std::sync::Arc;
+use std::time::Duration;
 use tokio::sync::broadcast::Receiver;
+use tokio::sync::Mutex;
+use tokio::task::{JoinError, JoinSet};
+use tokio::time::sleep as tokio_sleep;
+use tokio::time::Instant;
 use tracing::error;
+use tracing::log::info;
+use usecase::slots_collector::{SlotsCollector, SlotsGetter};
+
+pub async fn process_asset_details_stream_wrapper(
+    cloned_rx: Receiver<()>,
+    cloned_rocks_storage: Arc<Storage>,
+    last_saved_slot: u64,
+    first_processed_slot_value: u64,
+    gaped_data_client_clone: Client,
+) -> Result<(), JoinError> {
+    let processed_assets = process_asset_details_stream(
+        cloned_rx,
+        cloned_rocks_storage.clone(),
+        last_saved_slot,
+        first_processed_slot_value,
+        gaped_data_client_clone,
+    )
+    .await;
+
+    info!("Processed gaped assets: {}", processed_assets);
+
+    Ok(())
+}
+
+pub async fn process_raw_blocks_stream_wrapper(
+    cloned_rx: Receiver<()>,
+    cloned_rocks_storage: Arc<Storage>,
+    last_saved_slot: u64,
+    first_processed_slot_value: u64,
+    gaped_data_client_clone: Client,
+) -> Result<(), JoinError> {
```

Comment:
Those functions accept the same parameters. Is there any sense of creating a dedicated structure for them? Or it might be a single if-else function at all, with a flag `raw: bool `?